rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -0,0 +1,994 @@
+            RpcError::Cast("Response".to_owned(), "String".to_owned(), err.to_string())
+        })?;
+        serde_json::from_str(&response).map_err(|err| RpcError::RpcCall(err.to_string()))
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RpcResult<T> {
+    result: T,
+}
+
+#[derive(Debug)]
+pub struct TransactionTrace {
+    pub validate_invocation: RpcCallInfo,
+    pub function_invocation: RpcCallInfo,
+    pub fee_transfer_invocation: RpcCallInfo,
+    pub signature: Vec<StarkFelt>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
+pub struct RpcExecutionResources {
+    pub n_steps: usize,
+    pub n_memory_holes: usize,
+    pub builtin_instance_counter: HashMap<String, usize>,
+}
+
+#[derive(Debug)]
+pub struct RpcCallInfo {
+    pub execution_resources: VmExecutionResources,
+    pub retdata: Option<Vec<StarkFelt>>,
+    pub calldata: Option<Vec<StarkFelt>>,
+    pub internal_calls: Vec<RpcCallInfo>,
+}
+
+#[allow(unused)]
+#[derive(Debug, Deserialize)]
+pub struct RpcTransactionReceipt {
+    #[serde(deserialize_with = "actual_fee_deser")]
+    actual_fee: u128,
+    block_hash: StarkHash,
+    block_number: u64,
+    execution_status: String,
+    #[serde(rename = "type")]
+    tx_type: String,
+}
+
+fn actual_fee_deser<'de, D>(deserializer: D) -> Result<u128, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let hex: &str = Deserialize::deserialize(deserializer)?;
+    Ok(u128::from_str_radix(&hex[2..], 16).unwrap())
+}
+
+impl<'de> Deserialize<'de> for RpcCallInfo {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: serde_json::Value = Deserialize::deserialize(deserializer)?;
+
+        // Parse execution_resources
+        let execution_resources_value = value["execution_resources"].clone();
+
+        let execution_resources = VmExecutionResources {
+            n_steps: serde_json::from_value(execution_resources_value["n_steps"].clone())
+                .map_err(serde::de::Error::custom)?,
+            n_memory_holes: serde_json::from_value(
+                execution_resources_value["n_memory_holes"].clone(),
+            )
+            .map_err(serde::de::Error::custom)?,
+            builtin_instance_counter: serde_json::from_value(
+                execution_resources_value["builtin_instance_counter"].clone(),
+            )
+            .map_err(serde::de::Error::custom)?,
+        };
+
+        // Parse retdata
+        let retdata_value = value["result"].clone();
+        let retdata = serde_json::from_value(retdata_value).unwrap();
+
+        // Parse calldata
+        let calldata_value = value["calldata"].clone();
+        let calldata = serde_json::from_value(calldata_value).unwrap();
+
+        // Parse internal calls
+        let internal_calls_value = value["internal_calls"].clone();
+        let mut internal_calls = vec![];
+
+        for call in internal_calls_value.as_array().unwrap() {
```

Comment:
Errors were handled correctly up to this point. Are those `.unwrap()` intended?

Reply:
For now yes, the motivation of this state reader is to use it as a tool for debugging fee-related problems, which are the priority now. There are lots of `.unwrap()` which can all be solved in a future PR